pkg/clusterapi/aws/actuator.go

```diff
@@ -441,10 +441,22 @@ func (a *Actuator) updateStatus(machine *clusterv1.Machine, instance *ec2.Instan
 	mLog.Debug("updating status")
 
 	// Starting with a fresh status as we assume full control of it here.
-	awsStatus := &cov1.AWSMachineProviderStatus{}
+	awsStatus, err := controller.AWSMachineProviderStatusFromClusterAPIMachine(machine)
+	if err != nil {
+		return err
+	}
+	// Save this, we need to check if it changed later.
+	origInstanceID := awsStatus.InstanceID
 
 	// Instance may have existed but been deleted outside our control, clear it's status if so:
-	if instance != nil {
+	if instance == nil {
+		awsStatus.InstanceID = nil
```

Comment:
Can we add a test case to `TestUpdate` that verifies that these fields are being cleared when there is no longer an instance?

Reply:
This actually exposed that the code here was unreachable, we'd error out if no instances are found and not update status. Now updating status and then returning the error, test to match.